--- a/lib/heads_up_web/components/custom_components.ex
+++ b/lib/heads_up_web/components/custom_components.ex
@@ -15,4 +15,22 @@ defmodule HeadsUpWeb.CustomComponents do
     </div>
     """
   end
+
+  slot :inner_block, required: true
+  slot :tagline
+
+  def headline(assigns) do
+    assigns = assign(assigns, :emoji, ~w(👍 🙌 👊) |> Enum.random())
+
+    ~H"""
+    <div class="headline">
+      <h1>
+        <%= render_slot(@inner_block) %>
+      </h1>
+      <div :for={tagline <- @tagline} class="tagline">
+        <%= render_slot(tagline, @emoji) %>
+      </div>
+    </div>
+    """
+  end
 end
--- a/lib/heads_up_web/live/incident_live/index.ex
+++ b/lib/heads_up_web/live/incident_live/index.ex
@@ -12,6 +12,12 @@ defmodule HeadsUpWeb.IncidentLive.Index do
   def render(assigns) do
     ~H"""
     <div class="incident-index">
+      <.headline>
+        <.icon name="hero-trophy-mini" /> 25 Incidents Resolved This Month!
+        <:tagline :let={vibe}>
+          Thanks for pitching in. <%= vibe %>
+        </:tagline>
+      </.headline>
       <div class="incidents">
         <.incident_card :for={incident <- @incidents} incident={incident} />
       </div>
