--- a/lib/heads_up_web/router.ex
+++ b/lib/heads_up_web/router.ex
@@ -8,12 +8,23 @@ defmodule HeadsUpWeb.Router do
     plug :put_root_layout, html: {HeadsUpWeb.Layouts, :root}
     plug :protect_from_forgery
     plug :put_secure_browser_headers
+    plug :snoop
   end
 
   pipeline :api do
     plug :accepts, ["json"]
   end
 
+  def snoop(conn, _opts) do
+    answer = ~w(Yes No Maybe) |> Enum.random()
+
+    conn = assign(conn, :answer, answer)
+
+    IO.inspect(conn)
+
+    conn
+  end
+
   scope "/", HeadsUpWeb do
     pipe_through :browser
 
