--- a/lib/heads_up/incidents.ex
+++ b/lib/heads_up/incidents.ex
@@ -9,12 +9,41 @@ defmodule HeadsUp.Incidents do
 
   def filter_incidents(filter) do
     Incident
-    |> where(status: ^filter["status"])
-    |> where([i], ilike(i.name, ^"%#{filter["q"]}%"))
-    |> order_by(desc: :name)
+    |> with_status(filter["status"])
+    |> search_by(filter["q"])
+    |> sort(filter["sort_by"])
     |> Repo.all()
   end
 
+  defp with_status(query, status)
+       when status in ~w(pending resolved canceled) do
+    where(query, status: ^status)
+  end
+
+  defp with_status(query, _), do: query
+
+  defp search_by(query, q) when q in ["", nil], do: query
+
+  defp search_by(query, q) do
+    where(query, [i], ilike(i.name, ^"%#{q}%"))
+  end
+
+  defp sort(query, "name") do
+    order_by(query, :name)
+  end
+
+  defp sort(query, "priority_desc") do
+    order_by(query, desc: :priority)
+  end
+
+  defp sort(query, "priority_asc") do
+    order_by(query, asc: :priority)
+  end
+
+  defp sort(query, _) do
+    order_by(query, :id)
+  end
+
   def get_incident!(id) do
     Repo.get!(Incident, id)
   end
--- a/lib/heads_up_web/live/incident_live/index.ex
+++ b/lib/heads_up_web/live/incident_live/index.ex
@@ -27,6 +27,9 @@ defmodule HeadsUpWeb.IncidentLive.Index do
       <.filter_form form={@form} />
 
       <div class="incidents" id="incidents" phx-update="stream">
+        <div id="empty" class="no-results only:block hidden">
+          😢 No incidents found. Try changing your filters.
+        </div>
         <.incident_card
           :for={{dom_id, incident} <- @streams.incidents}
           incident={incident}
@@ -40,14 +43,23 @@ defmodule HeadsUpWeb.IncidentLive.Index do
   def filter_form(assigns) do
     ~H"""
     <.form for={@form} id="filter-form" phx-change="filter" phx-submit="filter">
-      <.input field={@form[:q]} placeholder="Search..." autocomplete="off" />
+      <.input field={@form[:q]} placeholder="Search..." autocomplete="off" phx-debounce="500" />
       <.input
         type="select"
         field={@form[:status]}
         prompt="Status"
         options={[:pending, :resolved, :canceled]}
       />
-      <.input type="select" field={@form[:sort_by]} prompt="Sort By" options={[:name, :priority]} />
+      <.input
+        type="select"
+        field={@form[:sort_by]}
+        prompt="Sort By"
+        options={[
+          Name: "name",
+          "Priority: High to Low": "priority_asc",
+          "Priority: Low to High": "priority_desc"
+        ]}
+      />
     </.form>
     """
   end
