--- a/lib/heads_up_web/live/incident_live/index.ex
+++ b/lib/heads_up_web/live/incident_live/index.ex
@@ -12,20 +12,36 @@ defmodule HeadsUpWeb.IncidentLive.Index do
     ~H"""
     <div class="incident-index">
       <div class="incidents">
-        <div :for={incident <- @incidents} class="card">
-          <img src={incident.image_path} />
-          <h2><%= incident.name %></h2>
-          <div class="details">
-            <div class="badge">
-              <%= incident.status %>
-            </div>
-            <div class="priority">
-              <%= incident.priority %>
-            </div>
-          </div>
+        <.incident_card :for={incident <- @incidents} incident={incident} />
+      </div>
+    </div>
+    """
+  end
+
+  attr :incident, HeadsUp.Incident, required: true
+
+  def incident_card(assigns) do
+    ~H"""
+    <div class="card">
+      <img src={@incident.image_path} />
+      <h2><%= @incident.name %></h2>
+      <div class="details">
+        <.badge status={@incident.status} />
+        <div class="priority">
+          <%= @incident.priority %>
         </div>
       </div>
     </div>
     """
   end
+
+  attr :status, :atom, values: [:pending, :resolved, :canceled], default: :pending
+
+  def badge(assigns) do
+    ~H"""
+    <div class="rounded-md px-2 py-1 text-xs font-medium uppercase inline-block border text-lime-600 border-lime-600">
+      <%= @status %>
+    </div>
+    """
+  end
 end
