--- a/lib/heads_up_web/live/incident_live/index.ex
+++ b/lib/heads_up_web/live/incident_live/index.ex
@@ -6,10 +6,18 @@ defmodule HeadsUpWeb.IncidentLive.Index do
 
   def mount(_params, _session, socket) do
     socket =
-      assign(socket,
-        incidents: Incidents.list_incidents(),
-        page_title: "Incidents"
-      )
+      socket
+      |> assign(:page_title, "Incidents")
+      |> stream(:incidents, Incidents.list_incidents())
+
+    IO.inspect(socket.assigns.streams.incidents, label: "mount")
+
+    socket =
+      attach_hook(socket, :log_stream, :after_render, fn
+        socket ->
+          IO.inspect(socket.assigns.streams.incidents, label: "AFTER RENDER")
+          socket
+      end)
 
     {:ok, socket}
   end
@@ -23,18 +31,23 @@ defmodule HeadsUpWeb.IncidentLive.Index do
           Thanks for pitching in. <%= vibe %>
         </:tagline>
       </.headline>
-      <div class="incidents">
-        <.incident_card :for={incident <- @incidents} incident={incident} />
+      <div class="incidents" id="incidents" phx-update="stream">
+        <.incident_card
+          :for={{dom_id, incident} <- @streams.incidents}
+          incident={incident}
+          id={dom_id}
+        />
       </div>
     </div>
     """
   end
 
   attr :incident, HeadsUp.Incidents.Incident, required: true
+  attr :id, :string, required: true
 
   def incident_card(assigns) do
     ~H"""
-    <.link navigate={~p"/incidents/#{@incident}"}>
+    <.link navigate={~p"/incidents/#{@incident}"} id={@id}>
       <div class="card">
         <img src={@incident.image_path} />
         <h2><%= @incident.name %></h2>
