--- a/lib/heads_up_web/live/effort_live.ex
+++ b/lib/heads_up_web/live/effort_live.ex
@@ -2,11 +2,15 @@ defmodule HeadsUpWeb.EffortLive do
   use HeadsUpWeb, :live_view
 
   def mount(_params, _session, socket) do
+    IO.inspect(self(), label: "MOUNT")
+
     socket = assign(socket, responders: 0, minutes_per_responder: 10)
     {:ok, socket}
   end
 
   def render(assigns) do
+    IO.inspect(self(), label: "RENDER")
+
     ~H"""
     <div class="effort">
       <h1>Community Love</h1>
@@ -31,6 +35,8 @@ defmodule HeadsUpWeb.EffortLive do
   end
 
   def handle_event("add", %{"quantity" => quantity}, socket) do
+    IO.inspect(self(), label: "ADD")
+
     socket = update(socket, :responders, &(&1 + String.to_integer(quantity)))
 
     {:noreply, socket}
