--- a/lib/heads_up_web/live/incident_live/index.ex
+++ b/lib/heads_up_web/live/incident_live/index.ex
@@ -13,7 +13,7 @@ defmodule HeadsUpWeb.IncidentLive.Index do
       socket
       |> assign(:page_title, "Incidents")
       |> assign(:form, to_form(params))
-      |> stream(:incidents, Incidents.filter_incidents(params))
+      |> stream(:incidents, Incidents.filter_incidents(params), reset: true)
 
     {:noreply, socket}
   end
@@ -64,7 +64,7 @@ defmodule HeadsUpWeb.IncidentLive.Index do
           "Priority: Low to High": "priority_desc"
         ]}
       />
-      <.link navigate={~p"/incidents"}>
+      <.link patch={~p"/incidents"}>
         Reset
       </.link>
     </.form>
@@ -97,7 +97,7 @@ defmodule HeadsUpWeb.IncidentLive.Index do
       |> Map.take(~w(q status sort_by))
       |> Map.reject(fn {_, v} -> v == "" end)
 
-    socket = push_navigate(socket, to: ~p"/incidents?#{params}")
+    socket = push_patch(socket, to: ~p"/incidents?#{params}")
 
     {:noreply, socket}
   end
