--- a/lib/heads_up_web/live/effort_live.ex
+++ b/lib/heads_up_web/live/effort_live.ex
@@ -11,6 +11,9 @@ defmodule HeadsUpWeb.EffortLive do
     <div class="effort">
       <h1>Community Love</h1>
       <section>
+        <button phx-click="add" phx-value-quantity="3">
+          + 3
+        </button>
         <div>
           <%= @responders %>
         </div>
@@ -26,4 +29,10 @@ defmodule HeadsUpWeb.EffortLive do
     </div>
     """
   end
+
+  def handle_event("add", %{"quantity" => quantity}, socket) do
+    socket = update(socket, :responders, &(&1 + String.to_integer(quantity)))
+
+    {:noreply, socket}
+  end
 end
